packages/admin-stories/src/docs/form/Overview.stories.mdx

```diff
@@ -1,15 +1,295 @@
+
+Final Form is strongly recommended if you
+
+-   are building an extensive or complicated form
+-   want to submit the form data to an API
+
+However, Final Form might not be sensible for every use case.
+Final Form is likely overkill if you want to create a solitary input field (e.g., a single select field for client-side filtering).
+In this case, you can use a plain HTML input combined with a React state.
+
+## `FinalForm` Component
+
+Comet Admin provides a `FinalForm` component.
+It's built on top of React Final Form's [`Form` component](https://final-form.org/docs/react-final-form/api/Form).
+
+There are two required properties:
+
+-   `mode` can be `add` (for creating a new resource) or `edit` (for editing an existing resource)
+-   `onSubmit()` handles the submission of form values to the API
+
+<Canvas>
+    <Story id="stories-form-finalform--basic-finalform" />
+</Canvas>
+
+### Submit Mutation Best Practices
+
+We recommend abiding by the following best practices considering form submission mutations:
+
+#### Location of GraphQL Mutations
+
+You should define form-related mutations in an adjacent file of the same name with the suffix `.gql.ts`.
+For example, if your form is located in the file `employees/EmployeeForm.tsx`, the mutation should be defined in `employees/EmployeeForm.gql.ts`.
+
+<Source
+    language="tsx"
+    code={dedent`
+        // employees/EmployeeForm.tsx
+        <FinalForm
+            mode="add"
+            onSubmit={(values) => {
+                // ...
+            }}
+        >
+            <Field label="Name" name="name" component={FinalFormInput} />
+            <SaveButton type="submit" />
+        </FinalForm>
+  `}
+/>
+
+<Source
+    language="ts"
+    code={dedent`
+        // employees/EmployeeForm.gql.ts
+        export const createEmployeeMutation = gql\`
+            mutation CreateEmployee($input: CreateEmployeeInput!) {
+                createEmployee(input: $input) {
+                    id
+                    name
+                }
+            }
+        \`;
+  `}
+/>
+
+If the form is short and simple, you can also define the mutation at the top of the same file.
+
+<Source
+    language="tsx"
+    code={dedent`
+        // employees/EmployeeForm.tsx
+        const createEmployeeMutation = gql\`
+            mutation CreateEmployee($input: CreateEmployeeInput!) {
+                createEmployee(input: $input) {
+                    id
+                    name
+                }
+            }
+        \`;
+        // ...
+        <FinalForm
+            mode="add"
+            onSubmit={(values) => {
+                // ...
+            }}
+        >
+            <Field label="Name" name="name" component={FinalFormInput} />
+            <SaveButton type="submit" />
+        </FinalForm>
+`}
+/>
+
+#### `apolloClient.mutate()` vs `useMutation()` Hook
+
+Prefer `apolloClient.mutate()` over the `useMutation()` hook to prevent the introduction of unnecessary new variables.
```

Comment:
The reasoning for this recommendation is not convincing. There are better arguments why someone probably doesn't need `useMutation`. For instance, `useMutation` offers additional `loading` and `error` states, which are not used, because Final Form manages those states itself (however, for this to work, the mutation needs to be wrapped in a try/catch statement and return `{ [FORM_ERROR]: ... }` upon error).

Reply:
Better like this?

> Prefer `apolloClient.mutate()` over the `useMutation()` hook.
> 
> The main advantage of `useMutation()` is that it provides access to the GraphQL loading and error states.
> However, the `FinalForm` component provides built-in handling for loading and error states.
> Custom handling is typically unnecessary.
> Therefore `apolloClient.mutate()` is sufficient for most use cases.
> 
> Also, `apolloClient.mutate()` helps you avoid a difficult variable naming process and keeps your code short and precise.

> however, for this to work, the mutation needs to be wrapped in a try/catch statement and return { [FORM_ERROR]: ... } upon error

This works out of the box because it's done directly in `FinalForm` (https://github.com/vivid-planet/comet-admin/blob/main/packages/admin/src/FinalForm.tsx#L209)